src/cpp/CMakeLists.txt

```diff
@@ -355,6 +361,110 @@ else()
    endif()
 endif()
 
+# SOCI
+if(UNIX)
+   set(RSTUDIO_TOOLS_SOCI "/opt/rstudio-tools/soci")
+   set(SOCI_INCLUDE_BUILD_DIR "${RSTUDIO_TOOLS_SOCI}/build/include")
+else()
+   set(RSTUDIO_TOOLS_SOCI "${RSTUDIO_WINDOWS_DEPENDENCIES_DIR}/install-soci/soci")
+   if(RSTUDIO_SESSION_WIN32)
+      set(SOCI_ARCH "x86")
+   else()
+      set(SOCI_ARCH "x64")
+   endif()
+   set(SOCI_INCLUDE_BUILD_DIR "${RSTUDIO_TOOLS_SOCI}/build/${SOCI_ARCH}/include")
+endif()
+
+set(SOCI_INCLUDE_DIR "${RSTUDIO_TOOLS_SOCI}/include")
+include_directories(SYSTEM ${SOCI_INCLUDE_DIR})
+include_directories(SYSTEM ${SOCI_INCLUDE_BUILD_DIR})
+
+# database library includes
+if (UNIX)
+   if (APPLE)
+      # We assume that libpq is available under the default brew install location
+      # since we install postgresql via brew in install-dependencies-osx
+      include_directories(SYSTEM "/usr/local/Cellar/postgresql/12.1/include")
+      include_directories(SYSTEM "/usr/local/Cellar/postgresql/12.1/include/libpq")
```

Comment:
This will break if the version of `postgresql` installed by Brew happens to change. You can access the current path using:

```
brew --prefix postgresql
```

Or, alternatively, Homebrew normally symlinks the current version of a formula to `/usr/local/opt`, so you could just check:

```
/usr/local/opt/postgresql
```

instead.

Aside: `libpq` and `postgresql` are separate formula; are you sure you want to use the `postgres` includes here or do you want to use `libpq` specifically?

Reply:
Because of how the includes are set up in SOCI, both of these paths are necessary, so this was intentional.